rust/connlib/tunnel/src/io.rs

```diff
@@ -179,16 +181,22 @@ impl Io {
         }
     }
 
-    pub fn rebind_dns_ipv4(&mut self, socket: SocketAddrV4) -> Result<()> {
-        self.udp_dns_server.rebind_ipv4(socket)?;
-        self.tcp_dns_server.rebind_ipv4(socket)?;
+    pub fn rebind_dns(&mut self, sockets: Vec<SocketAddr>) -> Result<()> {
+        tracing::debug!(?sockets, "Rebinding DNS servers");
 
-        Ok(())
-    }
+        self.udp_dns_server.clear();
+        self.tcp_dns_server.clear();
```

Comment:
Perhaps we could create a temp map with the new config and only assign it at the end of the function? otherwise, we could have partially broken config? 

Unless of course the intention is never recover from this and to always shut the tunnel.